app/assets/javascripts/validation.coffee

```diff
@@ -328,6 +334,74 @@ $(document).ready ->
     error.id = id
     error.element = $("##{id}")
 
+    # Hide individual required errors from an anyOf constraint
```

Comment:
Not saying this should be done now, but I think the below code should be refactored, the if/then/else nested structure is very deep makes it difficult to follow.

Reply:
I have tried to add comments to explain what decisions are being made and why. I thought about abstracting some of this into helper methods, but the need to return out of the function in some cases made it hard to figure out how to do that. If you have any suggestions I'm open to them. We can try to address them now or in the improvement ticket I mentioned.

Reply:
was using unless tried for readability on the cases where you just pass with ';'?

Reply:
It is a pretty nested set of if and else logic, and I think unless would make it much more complicated to understand.
Maybe I didn't explain it as well in the comment above, but I would have liked to abstract this logic into helper methods, but the need to return out of the function would necessitate adding more funkiness.
Also, because this is coffeescript, the `;` gets transpiled into `break;` to break out of the if/else logic. If it would be clearer, I can write it as `break;` instead.